### PR TITLE
Dynamically create a proxy.pac file

### PR DESCRIPTION
**Pull request by djones**
* Bonus bug fix where the heath check was actually returning "ok" text with a 500 HTTP response.
* Bonus readme updates.

This PR now also contains:
* #4 and #5

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -20,3 +60,24 @@ $ curl -x localhost:8080 http://proxyme.local/
 $ curl -x localhost:8080 http://fonts.googleapis.com/css?family=Titillium+Web:600,400,300,200
 ```
 
+## Running Jackproxy on Mac
+
+This is good for testing.
+
+```bash
+echo "{}" > proxymap.json
```
Comment (fotinakis) [inline README.md]: Drop this, just use the existing `testdata/proxymap.json`

File: README.md [hunk 2]
```diff
@@ -20,3 +60,24 @@ $ curl -x localhost:8080 http://proxyme.local/
...
+chmod +x jackproxy-darwin-386
+./jackproxy-darwin-386  --port 8080 --proxymap proxymap.json --source-hostname renderer.percy.local
```
Comment (fotinakis) [inline README.md]: At this layer we shouldn't reference renderer.percy.local, we just call it `proxyme.local` everywhere in jackproxy land.

File: jackproxy.go [hunk 3]
```diff
@@ -18,14 +18,30 @@ import (
 // Flags.
 var portFlag = flag.Int("port", 8080, "")
 var proxymapPathFlag = flag.String("proxymap", "", "JSON config file for mapping hijacked paths")
-var proxymeHostnameFlag = flag.String("proxyme-hostname", "proxyme.local", "")
+var sourceHostnameFlag = flag.String("source-hostname", "proxyme.local", "")
 
 // Healthcheck /healthz endpoint for the proxy itself.
 func healthzHandler(w http.ResponseWriter, req *http.Request) {
-	w.WriteHeader(http.StatusInternalServerError)
+	w.WriteHeader(http.StatusOK)
```
Comment (fotinakis) [inline jackproxy.go]: 🌮

File: README.md [hunk 4]
```diff
@@ -7,11 +7,51 @@ A custom hijacking reverse-proxy for Percy's internal rendering environment.
...
+For example I have this structure:
+
+```bash
+/Users/djones/Code/go/src/jackproxy
+```
+
+And I ran
+
+```bash
+$ export GOPATH=/Users/djones/Code/go/
```
Comment (djones) [inline README.md]: Will change these to `$HOME`

File: jackproxy.go [hunk 5]
```diff
@@ -18,14 +18,30 @@ import (
...
 	io.WriteString(w, "ok")
 }
 
+// Return a dynamic /proxy.pac file used for controlling when the proxy should be on or off
+func proxyPacHandler(w http.ResponseWriter, req *http.Request) {
+	w.WriteHeader(http.StatusOK)
+
+	// Pac file
+	var pacFile = fmt.Sprintf(
```
Comment (fotinakis) [inline jackproxy.go]: Maybe move this string to a constant at the top, and then still `fmt.Sprintf` it here?

File: jackproxy.go [hunk 6]
```diff
@@ -18,14 +18,30 @@ import (
...
-	w.WriteHeader(http.StatusInternalServerError)
+	w.WriteHeader(http.StatusOK)
 	io.WriteString(w, "ok")
 }
 
+// Return a dynamic /proxy.pac file used for controlling when the proxy should be on or off
+func proxyPacHandler(w http.ResponseWriter, req *http.Request) {
+	w.WriteHeader(http.StatusOK)
+
+	// Pac file
```
Comment (djones) [inline jackproxy.go]: This is kind of a useless comment, will remove it.

**Comment by fotinakis** [comment]
Back to you @djones

File: jackproxy.go (hunk 5 above)
Comment (djones) [reply jackproxy.go]: Gonna keep it there. I like the cohesion you get when it's inside the method like this.

File: README.md (hunk 1 above)
Comment (djones) [reply README.md]: Fixed.

File: testdata/proxymap.json [hunk 7]
```diff
@@ -1,14 +1,14 @@
 {
-  "http://proxyme.local/": {
+  "http://render.percy.local/": {
```
Comment (fotinakis) [inline testdata/proxymap.json]: These should remain `proxyme.local` for the purposes of this repo and to match the readme.

File: .gitignore [hunk 8]
```diff
@@ -25,3 +25,6 @@ _testmain.go
 *.exe
 *.test
 *.prof
+
+# Ignore this file used for testing
+proxymap.json
```
Comment (fotinakis) [inline .gitignore]: Oh also, drop this change?

**Pull request merged**